azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -385,6 +385,22 @@ private void uploadProjectFile(Connection connection, int projectId, int version
     updateChunksInProjectVersions(connection, projectId, version, chunks);
   }
 
+  public void addProjectVersion(
+      int projectId,
+      int version,
+      File localFile,
+      String uploader,
+      String uri) throws ProjectManagerException {
+    try (Connection connection = getConnection()) {
+      addProjectToProjectVersions(connection, projectId, version, localFile, uploader, uri);
+      connection.commit();
+    } catch (SQLException e) {
+      logger.error(e);
+      throw new ProjectManagerException(String.format("Add ProjectVersion failed. project id: %d version: %d",
+          projectId, version), e);
+    }
+  }
```

Comment:
I believe, in this case, you should explicitly call `DBUtils.close(connection)` here. Otherwise, you could call createQueryRunner(), then you don't need to worry about conn.commit, and conn.close. In future, we will use DatabaseOperator to replace.